api/src/main/resources/templates/pdfviewer/pdfviewer.vm

```diff
@@ -0,0 +1,157 @@
+          #if ($xcontext.userReference)
+            #set ($discard = 
+            	$digest.update($services.model.serialize($xcontext.userReference, 'default').getBytes('UTF-8')))
+          #end
+          #set ($discard = $digest.update($contentAuthor.getBytes('UTF-8')))
+          #set ($digest = $encoder.encode($digest.digest))
+          #set ($queryString = $escapetool.url({
+          	'reference': $services.model.serialize($document.documentReference, 'default'),
+            'filename': $attachment.filename,
+            'user': $contentAuthor,
+            'key': $digest
+          }))
+          #set ($url = $xwiki.getDocument('XWiki.PDFViewerService').getExternalURL('get', $queryString))
+        #end
+      #end
+    #end
+  #end
+#end
+
+#macro(displayFile $file)
+  #getUrlFromFile($file)
+  #if ($url)
+    #set ($pdfViewerConfigDoc = $xwiki.getDocument('PDFViewer.Code.PDFViewerConfiguration'))
+    #set ($trustedOrigins = $pdfViewerConfigDoc.getValue('trustedOrigins'))
+    #set ($withCredentials = $pdfViewerConfigDoc.getValue('withCredentials'))
+    #set ($trustedOrigins = $stringtool.join($trustedOrigins.split('\r\n'), ','))
+    #set ($queryString = $escapetool.url({
+      'file': $url,
+      'trustedOrigins': $trustedOrigins,
+      'withCredentials': $withCredentials
+    }))
+    #set ($pdfviewerurl = $services.webjars.url('pdf-js', 'web/viewer.html'))
+    ## Attachment URL have query parameters for cache issue.
+    #set ($pdfviewerurl = $pdfviewerurl.split('\?')[0])
+      <div>
+      <!--[if lt IE 10]>
+        <div class="box warningmessage">PDF Viewer : Not compatible with IE < 10</div>
+      <![endif]-->
+      <![if !(lt IE 10)]>
+        <iframe src="${pdfviewerurl}?${queryString}#locale=${xcontext.locale}" width="${width}" height="${height}">
+        </iframe>
+      <![endif]-->
+      </div>
+  #elseif (!$document || $document.isNew())
+    #error("$services.localization.render(
+    	'pdfviewer.error.nodocument', [$escapetool.html($escapetool.html($docname))])")
+  #else
+    #error("$services.localization.render('pdfviewer.error.noattachment',[$escapetool.html($filename),
+    	$escapetool.html($services.model.serialize($document.documentReference, 'default'))])")
+  #end
+#end
+
+#macro(isActive)
+  #set ($activeClass = $NULL)
+  #set ($isActive = $request.file == $file)
+  #if ("$!request.file" == '')
+    #set ($isActive = $foreach.index == 0)
+  #end
+#end
+
+#set ($mainPageReference = $services.model.createDocumentReference('', ['XWiki'], 'PDFViewerService'))
+#if (!$services.licensing.licensor.hasLicensureForEntity($mainPageReference))
+  #error("#getMissingLicenseMessage('pdfviewer.extension.name')")
+#else
+  #set ($fileParam = $file)
+  #set ($asAuthor = $asAuthor && ($asAuthor == 'true' || $asAuthor == 'yes' || $asAuthor == '1'))
+  #if (!$fileParam)
+    #error("$services.localization.render('pdfviewer.error.nofile')")
+  #else
+    #set ($files = $fileParam.split(','))
```

Comment:
So the file parameter supports multiple values? This means that a file name / reference can't a comma?

Reply:
Indeed
And I think this was added before the attachment picker was added. Right now you can add multiple values only by editing the source, since the selector won't let you select multiple values